tests/backup/backup_restore_basic_test.go

```diff
@@ -3842,3 +3843,333 @@ var _ = Describe("{KubeAndPxNamespacesSkipOnAllNSBackup}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
 	})
 })
+
+// ExcludeDirectoryFileBackup excludes mentioned directories or files from backed-up apps and restores them.
+var _ = Describe("{ExcludeDirectoryFileBackup}", func() {
+	var (
+		backupName           string
+		scheduledAppContexts []*scheduler.Context
+		AppContextsMapping   map[string]*scheduler.Context
+		namespace            string
+		bkpNamespaces        []string
+		backupNames          []string
+		restoreNames         []string
+
+		clusterUid        string
+		clusterStatus     api.ClusterInfo_StatusInfo_Status
+		restoreName       string
+		cloudCredName     string
+		cloudCredUID      string
+		backupLocationUID string
+		bkpLocationName   string
+
+		numDeployments int
+
+		providers         []string
+		backupLocationMap map[string]string
+		labelSelectors    map[string]string
+
+		storageClassExcludeFileDirMap map[string][]string
+		mountPathExcludeFileDirMap    map[string][]string
+		existingFileDirMountPathMap   map[string][]string
+		podMountPathMap               map[string][]string
+		fileListMountMap              map[string][]string
+		dirListMountMap               map[string][]string
+		formattedFileString           string
+		fileList                      []string
+		dirList                       []string
+		numOfFiles                    int
+		depthOfdirs                   int
+		levelOfdirs                   int
+		preRuleName                   string
+		postRuleName                  string
+		preRuleUid                    string
+		postRuleUid                   string
+	)
+	JustBeforeEach(func() {
+		backupNames = make([]string, 0)
+		restoreNames = make([]string, 0)
+		bkpNamespaces = make([]string, 0)
+		backupLocationMap = make(map[string]string)
+		labelSelectors = make(map[string]string)
+		AppContextsMapping = make(map[string]*scheduler.Context)
+		storageClassExcludeFileDirMap = make(map[string][]string)
+		mountPathExcludeFileDirMap = make(map[string][]string)
+		existingFileDirMountPathMap = make(map[string][]string)
+		podMountPathMap = make(map[string][]string)
+		fileListMountMap = make(map[string][]string)
+		dirListMountMap = make(map[string][]string)
+		numOfFiles = 5
+		depthOfdirs = 5
+		levelOfdirs = 0
+
+		numDeployments = 1
```

Comment:
Can you increase all these numbers?

Reply:
I know it is a sample test, but just want to check if the functions hold good with larger number